src/index.js

```diff
@@ -252,56 +259,65 @@ function UseWalletProvider({
         })
 
       if (!web3ReactConnector) {
-        throw new UnsupportedConnectorError(connectorId)
+        setStatus('error')
+        setError(new UnsupportedConnectorError(connectorId))
+        return
       }
 
       try {
         // TODO: there is no way to prevent an activation to complete, but we
         // could reconnect to the last provider the user tried to connect to.
-        setActivating(connectorId)
+        setConnector(connectorId)
         await web3ReactContext.activate(web3ReactConnector, null, true)
-        setActivated(connectorId)
-        setActivating(null)
+        setStatus('connected')
       } catch (err) {
-        setActivating(null)
-
         // Don’t throw if another connection has happened in the meantime.
         if (id !== activationId.current) {
           return
         }
+
+        // If not, the error has been thrown during the current connection attempt,
+        // so it's correct to indicate that there has been an error
+        setConnector(null)
+        setStatus('error')
+
         if (err instanceof UnsupportedChainIdError) {
-          throw new UnsupportedChainError(-1, chainId)
+          setError(new UnsupportedChainError(-1, chainId))
+          return
         }
         // It might have thrown with an error known by the connector
         if (connector.handleActivationError) {
-          connector.handleActivationError(err)
+          const thrownError = connector.handleActivationError(err)
+          setError(thrownError)
+          return
         }
-        // Otherwise, throw the received error
-        throw err
+
+        // Otherwise, set to state the received error
+        setError(err)
       }
     },
-    [chainId, connectors, deactivate, web3ReactContext]
+    [chainId, connectors, reset, web3ReactContext]
   )
 
   useEffect(() => {
     if (!account || !ethereum) {
+      setStatus('disconnected')
```

Comment:
Wondering if there was a reason to move this here rather than leaving it [in the `useEffect()` callback](https://github.com/aragon/use-wallet/pull/42/files#diff-1fdf421c05c1140f6d71444ea2b27638L304)? :thinking:

Reply:
Ah shoot; got confused here and thought there was a possible race condition we could have this case but not be `disconnected` in status. 😅